neo/SmartContract/Helper.cs

```diff
@@ -262,19 +262,10 @@ internal static bool VerifyWitnesses(this IVerifiable verifiable, Snapshot snaps
             if (hashes.Length != verifiable.Witnesses.Length) return false;
             for (int i = 0; i < hashes.Length; i++)
             {
-                byte[] verification = verifiable.Witnesses[i].VerificationScript;
-                if (verification.Length == 0)
-                {
-                    verification = snapshot.Contracts.TryGet(hashes[i])?.Script;
```

Comment:
Why we won't allow to get the verification from a hash? this could save a lot of space

Reply:
That is true, we just need to be sure that the scope inside that deployed "hash" will also be limited, otherwise, failed, right?

Reply:
Because in the next block, the deployed contract could be destroyed.

Reply:
vishhh...but is this a problem hard to solve, @erikzhang ?
If the verification pass and then it fails because of the tx's order on the block, is it really critical?

Reply:
Last time it was me removing this line Erik, and you reverted. Now it's time for me to ask you to revert this hahahah
As long as we maintain Storage.Get on verification, this is a very useful feature. Otherwise it will be more complicated to retrieve funds from contract as Sender.

Reply:
aehauheauheaueaahueahuea

Reply:
Transaction should have two main Scripts: one verification, which is executed for these things; and another for application only, processed together with chain state only. Another possibility is keep a single Script, but a flag on transaction indicating this is a Verification Only script. This would be the case for migrations/creation/destruction of contracts. 
A third and perhaps better possibility is to allow us to choose the trigger on tx Script invocation... so we could send a Tx with Verification Only, Application Only, or System Only, "Contract" Only perhaps other triggers that deal with contract, and are processed immediately on block proposal.

Reply:
I believe that we are going in the right direction, which will also match with the a pending proposal:

![image](https://user-images.githubusercontent.com/17395762/60044889-ca515000-9699-11e9-8673-2c392f7805e5.png)